### PR TITLE
Add oclif setup - Closes #554

### LINKED ISSUE #554: Change the library and change setup
### Description

Change the library and build settings.
Migrate most simple command for example.

### 
Child of #524

### PR DESCRIPTION
**Pull request by shuse2**
### Description
Change vorpal to oclif, and adding basic command and test for sample.

```
./bin/run warranty
./bin/run copyright
```

### Review checklist

* The PR solves #554 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-6.14.1
+8.9.4
```
Comment (willclarktech) [inline .nvmrc]: 8.11.3 is out

File: .babelrc [hunk 2]
```diff
@@ -1,5 +1,12 @@
 {
-	"presets": ["env"],
+	"presets": [
+    ["env", {
+      "targets": {
+        "node": "8"
+      }
+    }]
+  ],
+	"plugins": ["transform-object-rest-spread"],
```
Comment (willclarktech) [inline .babelrc]: node.green says this is available in node v8.3+: do we need a babel plugin?

File: .babelrc [hunk 3]
```diff
@@ -1,5 +1,12 @@
 {
-	"presets": ["env"],
+	"presets": [
+    ["env", {
+      "targets": {
+        "node": "8"
```
Comment (willclarktech) [inline .babelrc]: Maybe we should be more specific?

File: package.json [hunk 4]
```diff
@@ -20,64 +15,87 @@
 		"url": "https://github.com/LiskHQ/lisk-commander/issues"
 	},
 	"engines": {
-		"node": "=6.14.1",
-		"npm": "=3.10.10"
+		"node": ">=8.0.0"
```
Comment (willclarktech) [inline package.json]: Re object rest spread, we might want to increase this.

File: package.json [hunk 5]
```diff
@@ -20,64 +15,87 @@
...
 	},
```
Comment (willclarktech) [inline package.json]: Why not set the npm engine too? Especially if we'll start using a lockfile.

File: package.json [hunk 6]
```diff
@@ -20,64 +15,87 @@
...
+		"chai-as-promised": "7.1.1",
+		"coveralls": "3.0.0",
+		"eslint": "4.16.0",
+		"eslint-config-airbnb-base": "12.1.0",
+		"eslint-config-lisk-base": "1.0.0",
+		"eslint-config-oclif": "1.5.1",
+		"eslint-plugin-import": "2.8.0",
+		"eslint-plugin-mocha": "4.11.0",
+		"fancy-test": "1.2.0",
+		"globby": "8.0.1",
```
Comment (willclarktech) [inline package.json]: Do these new dependencies belong in this pr?

File: .babelrc [hunk 7]
```diff
@@ -1,5 +1,12 @@
 {
-	"presets": ["env"],
+	"presets": [
+    ["env", {
```
Comment (willclarktech) [inline .babelrc]: Indentation

File: bin/run [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env node
+
+require('@oclif/command').run()
+.then(require('@oclif/command/flush'))
```
Comment (willclarktech) [inline bin/run]: Indentation

File: bin/run.cmd [hunk 9]
```diff
@@ -0,0 +1,4 @@
+@echo off
```
Comment (willclarktech) [inline bin/run.cmd]: What's this file for?

Comment (shuse2) [reply bin/run.cmd]: this file is auto generated by oclif

File: bin/run (hunk 8 above)
Comment (shuse2) [reply bin/run]: this file is auto generated by oclif, but ill add it to the prettify and lint

File: package.json (hunk 6 above)
Comment (shuse2) [reply package.json]: This new dependencies were generated by oclif ='0

File: .babelrc (hunk 2 above)
Comment (shuse2) [reply .babelrc]: Ill changed the min version to 8.3+

File: .babelrc (hunk 3 above)
Comment (shuse2) [reply .babelrc]: Same as above, ill change to 8.3

File: bin/run.cmd (hunk 9 above)
Comment (willclarktech) [reply bin/run.cmd]: OK, what does it do though?

**Comment by shuse2** [comment]
@willclarktech 
In this PR, src and test are not that important.
Just added the sample command `warranty` and `copyright` without using config or tests.

Actual source code and test should be reviewed for the later PRs.

File: src/base.js [hunk 10]
```diff
@@ -13,14 +13,25 @@
...
  *
  */
-import { readJSONSync, writeJSONSync } from '../../../src/utils/fs';
+import { Command, flags as flagParser } from '@oclif/command';
 
-export function theJSONIsRead() {
-	const { filePath } = this.test.ctx;
-	this.test.ctx.returnValue = readJSONSync(filePath);
-}
+export default class BaseCommand extends Command {
```
Comment (willclarktech) [inline src/base.js]: Tests?

File: test/package.json [hunk 11]
```diff
@@ -0,0 +1,9 @@
+{
+	"name": "lisk-commander",
+	"oclif": {
+		"commands": "../src/commands",
+		"bin": "lisk",
+		"plugins": ["@oclif/plugin-help"]
+	},
+	"files": ["/bin", "/npm-shrinkwrap.json", "/oclif.manifest.json", "/dist"]
+}
```
Comment (willclarktech) [inline test/package.json]: Is this file supposed to be here?

File: test/setup.js [hunk 12]
```diff
@@ -13,14 +13,14 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (willclarktech) [inline test/setup.js]: Why is this necessary?

File: src/base.js (hunk 10 above)
Comment (shuse2) [reply src/base.js]: The test is added on the other PR since it needs more functionality to be added.

File: test/package.json (hunk 11 above)
Comment (shuse2) [reply test/package.json]: This is needed for changing the directory of tests target to `src`
Without this file, it will tests against `dist` folder, which is defined in the parent `package.json`

File: test/setup.js (hunk 12 above)
Comment (shuse2) [reply test/setup.js]: This is necessary because of the closest `package.json` in the test folder.
but maybe we can just add the dependencies on the `package.json` it self =(

File: src/index.js [hunk 13]
```diff
@@ -1,66 +1 @@
...
-
-if (process.env.NON_INTERACTIVE_MODE !== 'true') {
-	liskCommander.log(intro).show();
-}
-
-liskCommander.find('help').alias('?');
-liskCommander.find('exit').description(`Exits ${name}.`);
-
-export default liskCommander;
+export default '@oclif/command';
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/index.js]: This was one of the generated file ='(

File: src/commands/copyright.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+`.trim();
+
+export default class CopyrightCommand extends BaseCommand {
+	async run() {
+		this.log({ copyright });
+	}
+}
+
+CopyrightCommand.description = `
+Displays copyright notice.
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/copyright.js]: I think we need to raise the issue on `oclif`, it seems like the first line formatting doesn't work

File: package.json [hunk 15]
```diff
@@ -20,64 +15,85 @@
...
-		"coveralls": "=3.0.0",
-		"eslint": "=4.16.0",
-		"eslint-config-airbnb-base": "=12.1.0",
-		"eslint-config-lisk-base": "=1.0.0",
-		"eslint-plugin-import": "=2.8.0",
-		"eslint-plugin-mocha": "=4.11.0",
-		"husky": "=0.14.3",
-		"lint-staged": "=5.0.0",
-		"mocha": "=4.0.1",
-		"mocha-bdd": "=0.1.2",
```
[the remark this answers is not in the document]
Comment (shuse2) [reply package.json]: 😭

File: test/commands/copyright.test.js [hunk 16]
```diff
@@ -13,13 +13,15 @@
...
-	const returnValue = testFunction();
-	this.test.ctx.returnValue = returnValue;
-	return returnValue.catch(e => e);
-}
+describe('copyright command', () => {
+	test
+		.stdout()
+		.command(['copyright'])
+		.it('should show copy right', ctx => {
+			expect(ctx.stdout).to.contain(
```
[the remark this answers is not in the document]
Comment (shuse2) [reply test/commands/copyright.test.js]: Seems fancy test doesn't have it. it's using the mocha plugin as us on oclif

File: src/index.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply src/index.js]: Can we delete it though?

[the remark this answers is not in the document]
Comment (shuse2) [reply src/index.js]: I think we should keep it for now, im not entirely sure how the linux version is compiled
